server/src/main/java/org/apache/cassandra/sidecar/handlers/GossipUpdateHandler.java

```diff
@@ -0,0 +1,83 @@
+ * to you under the Apache License, Version 2.0 (the
+ * "License"); you may not use this file except in compliance
+ * with the License.  You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package org.apache.cassandra.sidecar.handlers;
+
+import java.util.Collections;
+import java.util.Set;
+
+import com.google.inject.Inject;
+import com.google.inject.Singleton;
+import io.vertx.core.http.HttpServerRequest;
+import io.vertx.core.net.SocketAddress;
+import io.vertx.ext.auth.authorization.Authorization;
+import io.vertx.ext.web.RoutingContext;
+import org.apache.cassandra.sidecar.acl.authorization.BasicPermissions;
+import org.apache.cassandra.sidecar.common.request.data.NodeCommandRequestPayload;
+import org.apache.cassandra.sidecar.common.server.StorageOperations;
+import org.apache.cassandra.sidecar.concurrent.ExecutorPools;
+import org.apache.cassandra.sidecar.utils.InstanceMetadataFetcher;
+import org.jetbrains.annotations.NotNull;
+
+import static org.apache.cassandra.sidecar.modules.HealthCheckModule.OK_STATUS;
+
+/**
+ * Handles {@code PUT /api/v1/cassandra/gossip} requests to start or stop Cassandra gossip.
+ *
+ * <p>Expects a JSON payload:
+ * { "state": "start" } or { "state": "stop" }
+ * and will asynchronously invoke the corresponding JMX operation.</p>
+ */
+@Singleton
+public class GossipUpdateHandler extends NodeCommandHandler implements AccessProtected
+{
+    @Inject
+    public GossipUpdateHandler(InstanceMetadataFetcher metadataFetcher, ExecutorPools executorPools)
+    {
+        super(metadataFetcher, executorPools, null);
+    }
+
+    @Override
+    public Set<Authorization> requiredAuthorizations()
+    {
+        return Collections.singleton(BasicPermissions.WRITE_GOSSIP.toAuthorization());
+    }
+
+    @Override
+    protected void handleInternal(RoutingContext context, HttpServerRequest httpRequest, @NotNull String host, SocketAddress remoteAddress, NodeCommandRequestPayload request)
+    {
+        StorageOperations storageOps = metadataFetcher.delegate(host).storageOperations();
+
+        executorPools.service().runBlocking(() -> {
+            switch (request.state())
+            {
+                case START:
+                    storageOps.startGossiping();
+                    break;
+                case STOP:
+                    storageOps.stopGossiping();
+                    break;
+                default:
+                    throw new IllegalStateException("Unknown state: " + request.state());
+            }
+        })
+                     .onSuccess(ignored -> context.json(OK_STATUS))
```

Comment:
I do not this this response is expected. 
`OK_STATUS` is used in the context of health check. It is defined in `HealthCheckModule`. 

I realized that the discussion in JIRA did not cover the response of the new APIs. My suggestion is to return 200 w/o body when the operation is successful.

Reply:
Talked Francisco offline. He exemplified where returning a non-empty response body could be useful for operators. 
I am updating my comment to either not use the `OK_STATUS` from `HealthCheckModule` or move the `OK_STATUS` (and `NOT_OK_STATUS`) to a place where it is more appropriate, say `ApiModule`.